contrib/babelfishpg_tds/src/include/tds_request.h

```diff
@@ -524,19 +524,21 @@ SetColMetadataForTvp(ParameterToken temp, const StringInfo message, uint64_t *of
 								"has a non-zero length database name specified. Database name is not allowed with a table-valued parameter, "
 								"only schema name and type name are valid.",
 								temp->paramOrdinal + 1, temp->paramMeta.colName.data, 1, 1, temp->type)));
-			initStringInfo(tempStringInfo);
+			initStringInfo(&tempStringInfo);
+			
+			memset(tempStringInfo.data, 0, tempStringInfo.maxlen);
```

Comment:
One of the reasons we use `StringInfoData` and `initStringInfo()` is to skip a low-level memory management like `memset()`. And PG uses `initStringInfo()` a lot without `memset()`. I'd like to know the true reason here :)

Reply:
Probably, the root cause is in `TdsUTF16toUTF8StringInfo()`?

Reply:
TdsUTF16toUTF8StringInfo is used in several other places as well, and it only uses the proper appendStringInfoChar() calls.

Reply:
The only place I can find this pointer `tempStringInfo.data` being used is in `temp->tvpInfo`. However the `temp` object is local to ReadParameters although the alloc'ed memory is not freed after use, so it seems on the surface unrelated to the stack that Valgrind complained about.